src/main/java/org/broadinstitute/consent/http/service/DacService.java

```diff
@@ -179,9 +183,22 @@ public void updateDac(String name, String description, String email, Integer dac
     dacDAO.updateDac(name, description, email, updateDate, dacId);
   }
 
-  public void deleteDac(Integer dacId) {
-    dacDAO.deleteDacMembers(dacId);
-    dacDAO.deleteDac(dacId);
+  public void deleteDac(Integer dacId) throws IllegalArgumentException, SQLException {
+    Dac fullDac = dacDAO.findById(dacId);
+    List<DataAccessAgreement> daas = daaDAO.findAll();
+    List<Dac> dacs = dacDAO.findAll();
+    Optional<Dac> broadDac = dacs.stream()
+        .filter(dac -> dac.getName().toLowerCase().contains("broad") && dac.getDacId().equals(dacId)).findFirst();
+    if (broadDac.isPresent()) {
+      throw new IllegalArgumentException("This is the Broad DAC, which can not be deleted.");
+    }
+    List<DataAccessAgreement> filteredDaas = daas.stream()
+        .filter(daa -> daa.getInitialDacId().equals(dacId) && ! daaService.isBroadDAA(daa.getDaaId(), daas, dacs)).toList();
+    try {
+      dacServiceDAO.deleteDacAndDaas(fullDac, filteredDaas);
```

Comment:
Due to the logic above, I don't think we need `filteredDaas` at all. We should be able to do all the necessary work with just the dac id.

Reply:
I think it's just a question of whether we want to filter out the DAAs not made by this DAC here, or in the ServiceDAO method.